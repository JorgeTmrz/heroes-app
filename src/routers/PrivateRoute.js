import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import PropTypes from 'prop-types'

export const PrivateRoute = ({
    isAuthenticated,
    component: Component,
    ...rest // the rest of the args passed to the component
}) => {

    const {location: {pathname: lastVisitedPage}} = rest;
    localStorage.setItem('lastPath', lastVisitedPage) // setting the las visited page on the localstorage

    return (
        <Route {...rest}
            component = {
                (props) => (
                    (isAuthenticated)
                    ? <Component {...props}/>
                    : <Redirect to = "/login" />
                )
            }
        />
    )
}

PrivateRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    Component: PropTypes.func.isRequired
}