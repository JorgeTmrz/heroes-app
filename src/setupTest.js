import "@testing-library/jest-dom";
import { configure } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
