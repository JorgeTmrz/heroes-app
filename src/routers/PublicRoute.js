import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import { DashboardRoutes } from './DashboardRoutes'
import { PrivateRoute } from './PrivateRoute'

export const PublicRoute = ({
    isAuthenticated,
    component: Component,
    ...rest

}) => {
    return (
        <Route {...rest}
            component = {(props) => (
                (isAuthenticated)
                ? <PrivateRoute 
                    isAuthenticated = {true}
                    component = {DashboardRoutes}
                />
                : <Component/>
            )}
        />
    )
}
