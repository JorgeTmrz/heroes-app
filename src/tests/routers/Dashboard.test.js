import "../../setupTest";
import { shallow } from "enzyme";
import { DashboardRoutes } from "../../routers/DashboardRoutes";

describe("Dashboard test cases", () => {
  test("should displayy itself correctly", () => {
    const wrapper = shallow(<DashboardRoutes />);

    expect(wrapper).toMatchSnapshot();
  });
});
