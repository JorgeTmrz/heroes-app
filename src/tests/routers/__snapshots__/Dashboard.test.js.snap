// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard test cases should displayy itself correctly 1`] = `
<Fragment>
  <Navbar />
  <div
    className="container mt-2"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/marvel"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/hero/:heroId"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/dc"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/search"
      />
      <Route
        component={[Function]}
        path="/"
      />
    </Switch>
  </div>
</Fragment>
`;
